src/treelearner/cuda_tree_learner.cpp

```diff
@@ -408,7 +408,7 @@ void CUDATreeLearner::copyDenseFeature() {
     // looking for dword_features_ non-sparse feature-groups
     if (!train_data_->IsMultiGroup(i)) {
       dense_feature_group_map_.push_back(i);
-      auto sizes_in_byte = train_data_->FeatureGroupSizesInByte(i);
+      auto sizes_in_byte = std::min(train_data_->FeatureGroupSizesInByte(i), static_cast<size_t>(num_data_));
```

Comment:
should we assert for the type for `FeatureGroupData` ? I think it should be 1-Byte type.

Reply:
I don't think an assert would be meaningful since FeatureGroupSizesInByte is returning a padded size (at least after 3.0.x it is).

Reply:
@guolinke Can we merge or this discussions should be resolved before that?

Reply:
I mean that using `static_cast<size_t>(num_data_)` as the size, the type should be 1-Byte type, right? 
FeatureGroup could be 4-bit, 1Byte, 2Byte and 4Byte. 

However, I think we can merge this first.